File: golang/patterns/creational/creational.go
package creational

import (
	"fmt"

	"github.com/Tamplier2911/gof-design-patterns/golang/patterns/creational/builder"
	"github.com/Tamplier2911/gof-design-patterns/golang/patterns/creational/factories"
	"github.com/Tamplier2911/gof-design-patterns/golang/patterns/creational/prototype"
	"github.com/Tamplier2911/gof-design-patterns/golang/patterns/creational/singleton"
)

func Creational() {
	fmt.Println("\nCreational")
	factories.Factories()
	builder.Builder()
	prototype.Prototype()
	singleton.Singleton()
}

File: csharp/pattenrs/Creational.cs
namespace Creational
{
    ///
    /// <summary>
    /// Class <c>CreationalPatterns</c> invokes all Creational patterns for testing purposes.
    /// Creational patterns: deal with creation (construction of objects)
    /// explicit (constructor) vs implicit (DI, reflection, etc...)
    /// wholesale (single statement) vs piecewise (step-by-step)
    /// </summary>
    ///
    class CreationalPatterns
    {
        public static void Run()
        {
            Console.WriteLine("\nCreational");

            Builder.Main.Run();
            Factories.Main.Run();
            Prototype.Main.Run();
            Singleton.Main.Run();
        }
    }
}
